File: packages/javascript/lib/verify/isUrl.ts
export function isUrl(url: string, withProtocol = false, strict = false): boolean {
	if (strict && url.indexOf('.') === -1) {
		return false;
	}

	try {
		new URL(`${withProtocol && url.indexOf('://') === -1 ? 'https://' : ''}${url}`).toString();
	} catch {
		return false;
	}

	return true;
}

File: packages/python/qsu/verify/isUrl.py
from urllib.parse import urlparse


def isUrl(url: str, withProtocol: bool = False, strict: bool = False) -> bool:
	if strict and url.find('.') == -1:
		return False

	target = (
		('https://' if withProtocol and url.find('://') == -1 else '') + url
	)

	try:
		parsed = urlparse(target)
	except ValueError:
		return False

	if not parsed.scheme or not parsed.netloc:
		return False

	return True
